archai/common/azureml_helper.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Dict, Union, Optional, Any
+from pathlib import Path
+
+from azure.ai.ml import MLClient
```

Comment:
Are users required to install this for main Archai? I think this should be installed only if pip install archai[az].

Reply:
Yes, I need to cleanup this mess.  Currently I'm doing this...
```
git clone git@github.com:microsoft/archai.git
cd archai
conda create -n sample-nas-env python=3.10
conda activate sample-nas-env
pip install -e .
cd docs\advanced_guide\cloud\azure\notebooks\multi_node_search
conda env update -f conda.yaml
pip install -r requirements.txt
```
The reason for the split between `conda.yaml` and `requirements.txt` is because the notebook uses more stuff that AML needs in the cloud.  The `conda.yaml` is also used in the cloud.

So I'll move this requirements.txt to a new `aml` section in the archai setup.py, does that good?  I'll also stop using `conda.yaml` locally and leave that one up to the AML cloud environment only, since it also installs archai from `https://github.com/microsoft/archai.git` and having 2 installs of archai in one python environment is confusing....

Reply:
Done